content/blog/dplyr-1-1-0-is-coming-soon/index.Rmd

```diff
@@ -0,0 +1,548 @@
+
+-   New join features, such as non-equi joins
+
+-   `arrange()` improvements with character vectors
+
+-   `morph()`, a generalization of `summarise()`
+
+-   Rewrites of "vector" functions using vctrs, including exciting updates to `case_when()`
+
+This pre-release blog post will discuss these new features in more detail.
+We feel pretty confident about them and can't wait for you to try them out, but we welcome any feedback on the [dplyr issues page](https://github.com/tidyverse/dplyr/issues).
+
+You can see a full list of changes in the [release notes](https://dplyr.tidyverse.org/dev/news/index.html).
+There are many additional improvements that couldn't fit in a single blog post!
+
+dplyr 1.1.0 is not on CRAN yet, but you can install the development version from GitHub with:
+
+```{r, eval = FALSE}
+pak::pak("tidyverse/dplyr")
+```
+
+The development version is mostly stable, but is still subject to minor changes before the official release.
+We don't encourage relying on it for production usage, but we would love for you to try out these new features.
+
+```{r setup, warning=FALSE, message=FALSE}
+library(dplyr)
+library(clock)
+set.seed(12345)
+```
+
+## Temporary grouping with `.by`
+
+Verbs that work "by group," such as `mutate()`, `summarise()`, `filter()`, and `slice()`, have gained a powerful new argument, `.by`, which allows for inline and temporary grouping.
+Grouping radically affects the computation of the dplyr verb you use it with, and one of the goals of `.by` is to allow you to place that grouping specification alongside the code that actually uses it.
+As an added benefit, with `.by` you no longer need to remember to `ungroup()` after `summarise()`, and `summarise()` won't ever message you about how it's handling the groups!
+
+This exciting feature was inspired by [data.table](https://cran.r-project.org/package=data.table), which bakes this idea directly into `[`.
+
+We'll explore `.by` with this `expenses` dataset, containing various `cost`s tracked across `id` and `region`.
+
+```{r}
+expenses <- tibble(
+  id = c(1, 2, 1, 3, 1, 2, 3),
+  region = c("A", "A", "A", "B", "B", "A", "A"),
+  cost = c(25, 20, 19, 12, 9, 6, 6)
+)
+expenses
+```
+
+If I were to ask you to compute the average `cost` per `region`, you'd probably write something like:
+
+```{r}
+expenses %>%
+  group_by(region) %>%
+  summarise(cost = mean(cost))
+```
+
+With `.by`, you can now write:
+
+```{r}
+expenses %>%
+  summarise(cost = mean(cost), .by = region)
+```
+
+These two particular results look the same, but the behavior of `.by` diverges from `group_by()` when multiple group columns are involved:
+
+-   `summarise(.by = )` always returns an ungrouped data frame
+
+-   `group_by() %>% summarise()` peels off 1 layer of grouping by default, and messages you that it is doing so unless you set `.groups`
+
+```{r}
+# Typically you'd follow this up with `ungroup()` or you'd set `.groups = "drop"`
```

Comment:
Probably don't need quite so much code here

Reply:
I think the example with multiple groups is important, so I've kept that, but I removed the bullets and generally simplified the language because I think the output tells most of the story